### ISSUE TITLE
java.lang.NoClassDefFoundError: nl/basjes/shaded/com/google/common/net/InternetDomainName

### ISSUE THREAD

**Issue by brwebb** [nielsbasjes/yauaa#106]
I experienced a NoClassDefFoundError exception when running yauaa.

my pom.xml

```
    <dependency>
      <groupId>nl.basjes.parse.useragent</groupId>
      <artifactId>yauaa</artifactId>
      <version>5.1</version>
    </dependency>
```

Stack trace

```
java.lang.NoClassDefFoundError: nl/basjes/shaded/com/google/common/net/InternetDomainName
        at nl.basjes.parse.useragent.UserAgentAnalyzerDirect.extractCompanyFromHostName(UserAgentAnalyzerDirect.java:888)
        at nl.basjes.parse.useragent.UserAgentAnalyzerDirect.determineDeviceBrand(UserAgentAnalyzerDirect.java:909)
        at nl.basjes.parse.useragent.UserAgentAnalyzerDirect.hardCodedPostProcessing(UserAgentAnalyzerDirect.java:874)
        at nl.basjes.parse.useragent.UserAgentAnalyzerDirect.parse(UserAgentAnalyzerDirect.java:764)
        at nl.basjes.parse.useragent.UserAgentAnalyzer.parse(UserAgentAnalyzer.java:78)
        at nl.basjes.parse.useragent.UserAgentAnalyzerDirect.parse(UserAgentAnalyzerDirect.java:711)
```

Here is the [line of code](https://github.com/nielsbasjes/yauaa/blob/v5.1/analyzer/src/main/java/nl/basjes/parse/useragent/UserAgentAnalyzerDirect.java#L888) throwing the exception.
`InternetDomainName domainName = InternetDomainName.from(hostname);`

I see that AgentAnalyzerDirect.java is [importing the appropriate library](https://github.com/nielsbasjes/yauaa/blob/v5.1/analyzer/src/main/java/nl/basjes/parse/useragent/UserAgentAnalyzerDirect.java#L20).
`import com.google.common.net.InternetDomainName;`

However, I don't see that package being imported in the [pom file](https://github.com/nielsbasjes/yauaa/blob/v5.1/analyzer/pom.xml).

Is this a case where the yauaa pom file is missing a dependency? Or am I using yauaa incorrectly?

Thanks!

**Comment by nielsbasjes** [comment]
Thank you.
I made a mistake in the shading part for Guava.
Fixed.

**Comment by nielsbasjes** [comment]
I released version 5.2 that fixes this serious issue.

**Comment by brwebb** [comment]
Thanks!
